src/main/java/slimeknights/tconstruct/plugin/jei/ToolRecipesCategory.java

```diff
@@ -0,0 +1,124 @@
+package slimeknights.tconstruct.plugin.jei;
+
+import com.mojang.blaze3d.vertex.PoseStack;
+import lombok.Getter;
+import mezz.jei.api.gui.builder.IRecipeLayoutBuilder;
+import mezz.jei.api.gui.drawable.IDrawable;
+import mezz.jei.api.gui.ingredient.IRecipeSlotsView;
+import mezz.jei.api.helpers.IGuiHelper;
+import mezz.jei.api.recipe.IFocusGroup;
+import mezz.jei.api.recipe.RecipeIngredientRole;
+import mezz.jei.api.recipe.RecipeType;
+import mezz.jei.api.recipe.category.IRecipeCategory;
+import net.minecraft.network.chat.Component;
+import net.minecraft.network.chat.MutableComponent;
+import net.minecraft.network.chat.contents.LiteralContents;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.world.item.ItemStack;
+import slimeknights.tconstruct.TConstruct;
+import slimeknights.tconstruct.library.client.GuiUtil;
+import slimeknights.tconstruct.library.recipe.tinkerstation.ToolRecipes;
+import slimeknights.tconstruct.library.tools.layout.LayoutSlot;
+import slimeknights.tconstruct.tools.TinkerTools;
+
+import javax.annotation.Nonnull;
+import java.util.Collections;
+import java.util.List;
+
+
+// IDK where to put this so im just leaving it the main jei directory
```

Comment:
These sorts of comments belong on the PR description or in a comment, not in the code.

Reply:
Where should I put it then?

Reply:
Doesn't matter.